pixiu/pkg/filter/prometheus/metric.go

```
@@ -0,0 +1,105 @@
+ */
+
+package prometheus
+
+import (
+	nh "net/http"
+	"time"
+)
+
+import (
+	"github.com/apache/dubbo-go-pixiu/pixiu/pkg/common/constant"
+	"github.com/apache/dubbo-go-pixiu/pixiu/pkg/common/extension/filter"
+	contextHttp "github.com/apache/dubbo-go-pixiu/pixiu/pkg/context/http"
+	"github.com/apache/dubbo-go-pixiu/pixiu/pkg/logger"
+	prom "github.com/apache/dubbo-go-pixiu/pkg/metrics/prometheus"
+)
+
+const (
+	Kind = constant.HTTPPrometheusMetricFilter
+)
+
+func init() {
+	filter.RegisterHttpFilter(&Plugin{})
+}
+
+type (
+	Plugin struct {
+	}
+
+	FilterFactory struct {
+		Cfg  *MetricCollectConfiguration
+		Prom *prom.Prometheus
+	}
+
+	Filter struct {
+		Cfg  *MetricCollectConfiguration
+		Prom *prom.Prometheus
+	}
+)
+
+func (p Plugin) Kind() string {
+	return Kind
+}
+
+func (p *Plugin) CreateFilterFactory() (filter.HttpFilterFactory, error) {
+
+	return &FilterFactory{
+		Prom: prom.NewPrometheus(),
+	}, nil
+}
+
+func (factory *FilterFactory) Config() interface{} {
+
+	return factory.Cfg
+}
+
+func (factory *FilterFactory) Apply() error {
+
+	return nil
+}
+
+func (factory *FilterFactory) PrepareFilterChain(ctx *contextHttp.HttpContext, chain filter.FilterChain) error {
+
+	f := &Filter{
+		Cfg:  factory.Cfg,
+		Prom: factory.Prom,
+	}
+	chain.AppendDecodeFilters(f)
+	return nil
+}
+
+func (f *Filter) Decode(ctx *contextHttp.HttpContext) filter.FilterStatus {
+
+	if f.Cfg == nil {
+		logger.Errorf("Message:Filter Metric Collect Configuration is null")
+		ctx.SendLocalReply(nh.StatusForbidden, constant.Default403Body)
+		return filter.Continue
+	}
+	if f.Prom == nil {
+		logger.Errorf("Message:Prometheus Collector is not initialized")
+		ctx.SendLocalReply(nh.StatusForbidden, constant.Default403Body)
+		return filter.Continue
+	}
+
+	f.Prom.SetMetricPath(f.Cfg.Rules.MeticPath)
+	f.Prom.SetPushGateway(f.Cfg.Rules.PushGatewayURL, time.Duration(f.Cfg.Rules.PushIntervalSeconds), f.Cfg.Rules.PushJobName)
+	start := f.Prom.HandlerFunc()
+	start(ctx)
```

should not ignore err